### PR TITLE
Added support for GeneralName (minus x400Address and ediPartyName) and GeneralNames

### PR DESCRIPTION
**Pull request by carl-wallace**

### CODE REVIEW

File: x509/src/general_name.rs [hunk 1]
```diff
@@ -0,0 +1,417 @@
+/// GeneralNames as defined in [RFC 5280 Section 4.2.1.6].
+use crate::alloc::string::ToString;
```
Comment (tarcieri) [inline x509/src/general_name.rs:2]: Toplevel doc comments should use `//!`

Also the `crate::` on the front of this import is redundant

```suggestion
//! GeneralNames as defined in [RFC 5280 Section 4.2.1.6].

use alloc::string::ToString;
```

File: x509/src/general_name.rs [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+}
+
+impl<'a> Sequence<'a> for OtherName<'a> {
+    fn fields<F, T>(&self, f: F) -> ::der::Result<T>
+    where
+        F: FnOnce(&[&dyn der::Encodable]) -> ::der::Result<T>,
+    {
+        f(&[&self.type_id, &self.value])
+    }
+}
```
Comment (tarcieri) [inline x509/src/general_name.rs:40]: Was there an issue deriving `Sequence` on this type?

File: x509/src/general_name.rs [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+
+/// GeneralNames as defined in [RFC 5280 Section 4.2.1.6] in support of the Subject Alternative Name extension.
+///
+/// ```text
+/// GeneralNames ::= SEQUENCE SIZE (1..MAX) OF GeneralName
+/// ```
+///
+/// [RFC 5280 Section 4.2.1.6]: https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.6
+pub type GeneralNames<'a> = alloc::vec::Vec<GeneralName<'a>>;
```
Comment (tarcieri) [inline x509/src/general_name.rs:49]: I'd suggest adding a `use` directive for `alloc::vec::Vec` at the top of the file

File: x509/src/general_name.rs [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+                let ia5 = decoder
+                    .context_specific::<Ia5String<'_>>(RFC822_NAME_TAG, TagMode::Implicit)?;
+                Ok(GeneralName::Rfc822Name(ia5.unwrap()))
+            }
+            0x82 => {
+                let ia5 =
+                    decoder.context_specific::<Ia5String<'_>>(DNS_NAME_TAG, TagMode::Implicit)?;
+                Ok(GeneralName::DnsName(ia5.unwrap()))
+            }
+            0xA3 => unimplemented!(),
```
Comment (tarcieri) [inline x509/src/general_name.rs:125]: Are these a TODO? I think it's bad to panic as opposed to returning an error.

File: x509/src/general_name.rs (hunk 1 above)
Comment (carl-wallace) [reply x509/src/general_name.rs:2]: Fixed.

File: x509/src/general_name.rs (hunk 2 above)
Comment (carl-wallace) [reply x509/src/general_name.rs:40]: Yes. If I comment out the above and add Sequence back to the derive, the following results:
```
the trait `DecodeValue<'_>` is not implemented for `OtherName<'_>`
```

File: x509/src/general_name.rs (hunk 3 above)
Comment (carl-wallace) [reply x509/src/general_name.rs:49]: Fixed.

File: x509/src/general_name.rs (hunk 2 above)
Comment (tarcieri) [reply x509/src/general_name.rs:40]: Weird. Guess I'll have to take a look at the `cargo expand` output and see why the derived implementation differs, but I can do that after merging this.

File: x509/src/general_name.rs (hunk 4 above)
Comment (carl-wallace) [reply x509/src/general_name.rs:125]: Bah. Yes, those should be errors. I see no reason to mark as TODO but forgot to fix that. FIxed now.

**Pull request merged**